client_middleware.go

```diff
@@ -85,7 +85,11 @@ func (g *gzipClientMiddleware) ClientMiddleware(next client.Endpoint) client.End
 			return
```

Comment:
Welcome to contribute to this middleware, can you help me return the `error` here?